### ISSUE TITLE
Check the cost function MNIST

### ISSUE THREAD

**Issue by mkao006** [mkao006/deep_learning#5]

**Comment by mkao006** [comment]
Currently using cross-entropy. However, it is causing too many numerical problems. Maybe we should switch to softmax

**Comment by mkao006** [comment]
The cross-entropy works fine now, it was the specification of the derivative of cost that was wrong.
